test/src/test/java/org/apache/accumulo/test/functional/TableChangeStateIT.java

```diff
@@ -0,0 +1,549 @@
+        status.setError(ex);
+      }
+
+    }
+
+    /**
+     * Provide OnlineStatus of this operation to determine when complete and timing information
+     *
+     * @return OnlineStatus to determine when complete and timing information
+     */
+    public OnlineStatus getStatus() {
+      return status;
+    }
+
+  }
+
+  /**
+   * Instance to create / run a compaction using a slow iterator.
+   */
+  private class SlowCompactionRunner implements Runnable {
+
+    private final String tableName;
+
+    /**
+     * Create an instance of this class.
+     *
+     * @param tableName
+     *          the name of the table that will be compacted with the slow iterator.
+     */
+    SlowCompactionRunner(final String tableName) {
+      this.tableName = tableName;
+    }
+
+    @Override
+    public void run() {
+
+      long startTimestamp = System.nanoTime();
+
+      IteratorSetting slow = new IteratorSetting(30, "slow", SlowIterator.class);
+      SlowIterator.setSleepTime(slow, 100);
+
+      List<IteratorSetting> compactIterators = new ArrayList<>();
+      compactIterators.add(slow);
+
+      log.trace("Slow iterator {}", slow.toString());
+
+      try {
+
+        Connector connector = getConnector();
+
+        log.trace("Start compaction");
+
+        connector.tableOperations().compact(tableName, new Text("0"), new Text("z"), compactIterators, true, true);
+
+        log.trace("Compaction wait is complete");
+
+        log.trace("Slow compaction of {} rows took {} ms", NUM_ROWS, TimeUnit.MILLISECONDS.convert((System.nanoTime() - startTimestamp), TimeUnit.NANOSECONDS));
+
+        // validate that number of rows matches expected.
+
+        startTimestamp = System.nanoTime();
+
+        // validate expected data created and exists in table.
+
+        Scanner scanner = connector.createScanner(tableName, Authorizations.EMPTY);
+
+        int count = 0;
+        for (Map.Entry<Key,Value> elt : scanner) {
+          String expected = String.format("%05d", count);
+          assert (elt.getKey().getRow().toString().equals(expected));
+          count++;
+        }
+
+        log.trace("After compaction, scan time for {} rows {} ms", NUM_ROWS,
+            TimeUnit.MILLISECONDS.convert((System.nanoTime() - startTimestamp), TimeUnit.NANOSECONDS));
+
+        if (count != NUM_ROWS) {
+          throw new IllegalStateException(String.format("After compaction, number of rows %1$d does not match expected %2$d", count, NUM_ROWS));
+        }
+
+      } catch (TableNotFoundException ex) {
+        throw new IllegalStateException("test failed, table " + tableName + " does not exist", ex);
+      } catch (AccumuloSecurityException ex) {
+        throw new IllegalStateException("test failed, could not add iterator due to security exception", ex);
+      } catch (AccumuloException ex) {
+        // test cancels compaction on complete, so ignore it as an exception.
+        if (!ex.getMessage().contains("Compaction canceled")) {
```

Comment:
This stack element is thrown: 

Caused by: org.apache.accumulo.core.client.AccumuloException: Compaction canceled.  

Did not want to swallow all AccumuloExceptions - just the Compaction cancel. (Not sure why this is an exception when requested, but it probably is a rare event and does provide history in the log.)